src/java_bytecode/generate_java_generic_type.cpp

```diff
@@ -126,46 +126,3 @@ irep_idt generate_java_generic_typet::build_generic_tag(
 
   return new_tag_buffer.str();
 }
-
-
-/// Activate the generic instantiation code.
-/// \param message_handler
-/// \param symbol_table The symbol table so far.
-void
-instantiate_generics(
-  message_handlert &message_handler,
-  symbol_tablet &symbol_table)
-{
-  generate_java_generic_typet instantiate_generic_type(message_handler);
-  // check out the symbols in the symbol table at this point to see if we
-  // have a a generic type in.
-  for(const auto &symbol : symbol_table.symbols)
-  {
-    if(symbol.second.type.id()==ID_struct)
-    {
-      auto symbol_struct=to_struct_type(symbol.second.type);
-      auto &components=symbol_struct.components();
-
-      for(const auto &component : components)
-      {
-        if(is_java_generic_type(component.type()))
-        {
-          const auto &type_vars=to_java_generic_type(component.type()).
-            generic_type_variables();
-
-          // Before we can instantiate a generic component, we need
-          // its type variables to be instantiated parameters
-          if(all_of(type_vars.cbegin(), type_vars.cend(),
```

Comment:
@NlightNFotis This is your old code for checking that all generic parameters are instantiated - you may wish to reuse this in your `smart_select_pointer_typet`